app/models/service.rb

```diff
@@ -235,6 +235,10 @@ def backend_authentication_value
     end
   end
 
+  def independent_mapping_rules?
+    act_as_product? || provider_can_use?(:independent_mapping_rules)
```

Comment:
Well we do not need this rolling update as it is only for 2.5 patch
On master we can just roll it out for API as Product